File: UPCRideAndroid/app/src/main/java/com/example/upcride/activities/ViajesFragment.java
package com.example.upcride.activities;


import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.upcride.R;

import java.util.ArrayList;
import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 */
public class ViajesFragment extends Fragment {

    private RecyclerView recyclerViewViaje;
    private ViajesAdapterHistory adaptadorViaje;

    public ViajesFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View vista = inflater.inflate(R.layout.fragment_viajes, container, false);

        recyclerViewViaje = (RecyclerView) vista.findViewById(R.id.trips_recycler);
        recyclerViewViaje.setLayoutManager(new LinearLayoutManager(getContext()));

        adaptadorViaje = new ViajesAdapterHistory(listaViajes());
        recyclerViewViaje.setAdapter(adaptadorViaje);


        return vista;
    }

    public List<ViajeModelo> listaViajes(){
        List<ViajeModelo> viajes = new ArrayList<>();
        viajes.add(new ViajeModelo("Brian", "09:10", "Nuevo viaje de mi casa a la u", "Av.Izaguirre", "UPC San Miguel", "20", "10", R.drawable.ricardo));
        viajes.add(new ViajeModelo("Sebastian", "22:10", "Mensaje de prueba", "Carabayllo", "UPC San Monterrioc", "20", "10", R.drawable.queso));
        viajes.add(new ViajeModelo("Emilio", "21:10", "Muchachos estare realizando un viaje por toda Universitaria", "Puente Piedra", "UPC San Isidro", "20", "10", R.drawable.calamardo));
        viajes.add(new ViajeModelo("Brian", "09:10", "Nuevo viaje para probar", "Comas", "UPC San Miguel", "20", "10", R.drawable.ricardo));
        return viajes;
    }

}

File: UPCRideKotlin/app/src/main/java/com/example/upcridekotlin/activities/ViajesFragment.kt
package com.example.upcridekotlin.activities

import android.os.Bundle

import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.upcridekotlin.R
import com.example.upcridekotlin.model.ViajeModelo

import java.util.ArrayList

/**
 * A simple [Fragment] subclass.
 */
class ViajesFragment : Fragment() {

    private var recyclerViewViaje: RecyclerView? = null
    private var adaptadorViaje: ViajesAdapterHistory? = null


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val vista = inflater.inflate(R.layout.fragment_viajes, container, false)

        recyclerViewViaje = vista.findViewById(R.id.trips_recycler)
        recyclerViewViaje!!.layoutManager = LinearLayoutManager(context)

        adaptadorViaje = ViajesAdapterHistory(listaViajes())
        recyclerViewViaje!!.adapter = adaptadorViaje


        return vista
    }

    fun listaViajes(): List<ViajeModelo> {
        val viajes = ArrayList<ViajeModelo>()
        viajes.add(
            ViajeModelo(
                10,
                "Brian",
                "09:10",
                "Nuevo viaje",
                "Av. Sucre 111",
                "UPC Villa",
                "20",
                "10",
                R.drawable.ricardo
            )
        )
        viajes.add(
            ViajeModelo(
                10,
                "Sebastian",
                "22:10",
                "Nuevo viaje",
                "Av. La Marina 344",
                "UPC Monterrico",
                "20",
                "10",
                R.drawable.queso
            )
        )
        viajes.add(
            ViajeModelo(10,
                "Emilio",
                "21:10",
                "Muchachos, les tengo un viaje",
                "Av. Izaguirre 999",
                "UPC San Miguel",
                "20",
                "10",
                R.drawable.calamardo
            )
        )
        viajes.add(
            ViajeModelo(10,
                "Brian",
                "09:10",
                "Nuevo viaje",
                "Av. Sucre 111",
                "UPC Villa",
                "20",
                "10",
                R.drawable.ricardo
            )
        )
        return viajes
    }

}// Required empty public constructor
